modules/contracts/apps/payment/src/handlers/execute.rs

```diff
@@ -79,6 +86,11 @@ pub fn tip(
 
     // Search for trading pairs between the deposited assets and the desired asset
     for pay_asset in asset_entries {
+        // No need to swap if desired asset sent
+        if pay_asset.name == desired_asset {
+            total_amount += pay_asset.amount;
+            continue;
+        }
         // query the pools that contain the desired asset
         let query_msg = ans_host::QueryMsg::PoolList {
```

Comment:
We should have this as a function on `AnsHost` that accepts an optional filter and does the smart-query.

Reply:
> We should have this as a function on `AnsHost` that accepts an optional filter and does the smart-query.

We have `pool_list()` method already, updated to use that instead